wit/types.wit

```diff
@@ -1,44 +1,43 @@
+        timeout,
+        /// An error occurred with the connection. Includes a message for additional context
+        connection(string),
+        /// A permission error occurred. Includes a message for additional context
+        permission-denied(string),
+        /// A catch all for other types of errors
+        other(string),
     }
   
-    /// There are two types of channels:
-    /// - publish-subscribe channel, which is a broadcast channel, and
-    /// - point-to-point channel, which is a unicast channel.
-    ///
-    /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
-    type channel = string;
-  
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
-    record guest-configuration {
-        channels: list<channel>,
-        extensions: option<list<tuple<string, string>>>
-    }
-  
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
-    record message {
-        data: list<u8>,
-        format: format-spec,
-        metadata: option<list<tuple<string, string>>>
+    /// A message with a binary payload and additional information
+    resource message {
+        constructor(topic: string, data: list<u8>);
+        /// The topic/subject/channel this message was received or should be sent on
+        topic: func() -> string;
+        /// Set the topic/subject/channel this message should be sent on
+        set-topic: func(topic: string);
+        /// An optional content-type describing the format of the data in the message. This is 
+        /// sometimes described as the "format" type
+        content-type: func() -> option<string>;
+        /// Set the content-type describing the format of the data in the message. This is
+        /// sometimes described as the "format" type
+        set-content-type: func(content-type: string);
+        /// An opaque blob of data
+        data: func() -> list<u8>;
+        /// Set the opaque blob of data for this message, discarding the old value
+        set-data: func(data: list<u8>);
+        /// Optional metadata (also called headers or attributes in some systems) attached to the
+        /// message
+        metadata: func() -> option<list<tuple<string, string>>>;
+        /// Add a new key-value pair to the metadata, overwriting any existing value for the same key
+        add-metadata: func(key: string, value: string);
```

Comment:
Should `set-metadata` also be available to "reset" the entired metadata?

Reply:
Or, `set-metadata: func(metadata: option<list<tuple<string, string>>>)` could replace `add-metadata`

Reply:
yeah replace, reset, set ... ambiguous language, I should use replace word 😄

Reply:
Added a `remove-metadata` function that should address this use-case ✅